### ISSUE TITLE
Extreme performance drop since 5.6

### ISSUE THREAD

**Issue by koisama** [transmission-remote-gui/transgui#1163]
I run instances of transmission-daemon that seed quite a few torrents, test instances had 17k and 56k.
The smaller 17k instance needs about 1 second of CPU time to load a list of torrents on 5.5.1. On 5.6 it needs 47 seconds of CPU time. Bigger 56k instance requires 5 seconds of CPU time with 5.5.1, and more than 10 minutes on 5.6. 

I'm running it on windows 7 with transmission-daemon 2.93. 
The issue is present in 5.6, 5.10, 5.13, 5.15.3, 5.15.4.
Versions 5.5.1, 5.4, 5.0.1 all work.
When built from source, 5.5.1 works as expected, 5.15.4 is extremely slow.

The culprit seems to be in main.pas:7157, repeated calls to FTorrents.IndexOf(idxTorrentId, ids[i]).
Optimizing it - calling FTorrents.IndexOf(idxTorrentId, ids[i]) once and using saved id later reduces loading time to ~1:50, but it's still too slow to be considered even remotely acceptable.

**Comment by antekgla** [comment]
Try to un-check **View --> Status Bar Sizes**

**Comment by koisama** [comment]
Oh, there was an option.
Unchecking **Status Bar Sizes** indeed helps.

**Comment by antekgla** [comment]
Sorry have to comment...
You have a Transmission running 17.000 and 56.000 torrents????
...Wow  :exclamation:

**Comment by koisama** [comment]
Yes, it's a little bit more than people would reasonably expect...

Maybe use hash maps? I tested with GContnrs hashmap and the performance was acceptable.

**Comment by satter** [comment]
I'm using Transmission remote GUI 5.15.4 on MacOS with Transmission daemon 2.84 on Ubuntu server. Now I have 687 torrents and GUI responds too slow. When I had over 700 torrents GUI was barely usable. Is it possible to tune some parameters for better performance?

**Comment by satter** [comment]
@antekgla sorry, I did not mention that **Status Bar Sizes** option does not affect performance in my case.

**Comment by leonsoft-kras** [comment]
Try it  https://yadi.sk/d/8psrAQ0tngmQUw
Report the result please.

**Comment by satter** [comment]
> Try it https://yadi.sk/d/8psrAQ0tngmQUw
> Report the result please.

I have issues with MacOS version. Windows app is totally fine.
There also was 32bit deprecation notice on first launch after OS upgrade to Mojave.

**Comment by PeterDaveHello** [comment]
macOS 32bit version is another issue.

Anyway, I've increased the optimization level in v5.16.0, hope it might help:
https://github.com/transmission-remote-gui/transgui/releases/tag/v5.16.0

**Comment by satter** [comment]
@PeterDaveHello I've installed 5.16 from brew, it is not buttery-smooth but a way much better than 5.15.4
thanks!

**Comment by PeterDaveHello** [comment]
@satter Glad it helps!
